MISRA.md

```diff
@@ -21,3 +21,13 @@ _Ref 5.4.1_
         in use and changing the name will break existing user projects. Thus, for
         backwards compatibility, the macro is not modified and kept as is and the
         deviation is suppressed.
+
+_Ref 18.2.1_
+
+- MISRA Rule 18.2 states that two pointers may only be subtracted if they point
```

Comment:
Since we have offsets, this should not require pointer arithmetic? Could we convert this to do the math on indices?

Reply:
Please reference [this reply](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk/pull/92#discussion_r1510667278).